examples/full.rs

```diff
@@ -0,0 +1,416 @@
+use std::{
+    ffi::{c_int, c_void},
+    fs::{File, OpenOptions},
+    io::Write,
+    os::fd::AsRawFd,
+    ptr,
+    sync::Arc,
+};
+
+use cudarc::driver::{
+    sys::{
+        cuExternalMemoryGetMappedBuffer,
+        cuImportExternalMemory,
+        CUDA_EXTERNAL_MEMORY_HANDLE_DESC_st__bindgen_ty_1,
+        CUdeviceptr,
+        CUexternalMemory,
+        CUexternalMemoryHandleType,
+        CUresult,
+        CUDA_EXTERNAL_MEMORY_BUFFER_DESC,
+        CUDA_EXTERNAL_MEMORY_HANDLE_DESC,
+    },
+    CudaDevice,
+};
+#[allow(deprecated)]
+use nvidia_video_codec_sdk::sys::nvEncodeAPI::NV_ENC_PRESET_LOW_LATENCY_HP_GUID;
+use nvidia_video_codec_sdk::{
+    safe::{api::ENCODE_API, buffer::MappedResource, encoder::Encoder},
+    sys::nvEncodeAPI::{
+        NV_ENC_BUFFER_FORMAT::NV_ENC_BUFFER_FORMAT_ARGB,
+        NV_ENC_CODEC_H264_GUID,
+        NV_ENC_H264_PROFILE_HIGH_GUID,
+        NV_ENC_INITIALIZE_PARAMS,
+        NV_ENC_INPUT_RESOURCE_TYPE,
+        NV_ENC_PIC_PARAMS,
+        NV_ENC_PIC_STRUCT,
+        NV_ENC_REGISTER_RESOURCE,
+        NV_ENC_TUNING_INFO,
+        _NV_ENC_BUFFER_FORMAT,
+    },
+};
+use vulkano::{
+    device::{
+        physical::PhysicalDeviceType,
+        Device,
+        DeviceCreateInfo,
+        DeviceExtensions,
+        QueueCreateInfo,
+    },
+    instance::{Instance, InstanceCreateInfo},
+    memory::{
+        DeviceMemory,
+        ExternalMemoryHandleType,
+        ExternalMemoryHandleTypes,
+        MappedDeviceMemory,
+        MemoryAllocateInfo,
+        MemoryPropertyFlags,
+    },
+    VulkanLibrary,
+};
+
+/// Returns the color `(r, g, b, alpha)` of a pixel on the screen relative to
+/// its position on a screen:
+///
+/// Top right will be red,
+/// bottom left will be green,
+/// all colours will shift towards having more blue as `time` increases.
+///
+/// # Arguments
+///
+/// * `width`, `height` - Width and height of the screen.
+///
+/// * `x`, `y` - Coördinates of the pixel on the screen.
+///
+/// * time - Fraction indicating what part of the animation we are in [0,1]
```

Comment:
Can you remove the empty lines between each bullet point?
```suggestion
/// * `width`, `height` - Width and height of the screen.
/// * `x`, `y` - Coördinates of the pixel on the screen.
/// * time - Fraction indicating what part of the animation we are in [0,1]
```